cg/meta/archive/archive.py

```diff
@@ -17,6 +19,14 @@ class FileAndSample(BaseModel):
     sample: Sample
 
 
+class ArchiveModels(BaseModel):
```

Comment:
I think the current approach violates the Open/Closed Principle since the code in this PR will need modification (?) if we add more ArchiveLocationsInUse. Ideally, the logic associated with each ArchiveLocationInUse should be encapsulated separately.

Reply:
Maybe we already do this with the `handler_map` 😅

Reply:
Hmm, in what way are you thinking @seallard ? model_config, file_model and handler is intended to keep all the implementation specific data, wherein file_model specifies what data is needed to archive in the specified flow and "handler" is used to invoke that flow. 

I have a hard time seeing an interface with less data than this.

Reply:
Oh wrote before your second comment 😅 Yes, handler_map is intended for this but I agree that the current placement might not be optimal.